grouper/queue.py

```diff
@@ -102,24 +102,36 @@ def push_advisor_msg(self, org_id: str, inventory_id: str, advisor_changed: bool
             item.second_upload_event.set()
         elif not is_updated:
             QUEUE_SIZE.inc()
-            create_task_and_log(self._start_item_processing(org_id, inventory_id), LOGGER, self.loop)
+            create_task_and_log(self._start_item_processing(org_id, inventory_id, reporter), LOGGER, self.loop)
 
-    async def _start_item_processing(self, org_id: str, inventory_id: str):
+    async def _start_item_processing(self, org_id: str, inventory_id: str, reporter: str):
         """Single queue item waiting coroutine"""
         item = self._queue[inventory_id]
-        LOGGER.debug(
-            "starting waiting for %s msg for system: %s, account: %s",
-            "inventory" if item.advisor_upload else "advisor",
-            inventory_id,
-            org_id,
-        )
 
-        try:
-            await asyncio.wait_for(item.second_upload_event.wait(), CFG.grouper_messages_timeout_sec)
-            PAIR_HIT.inc()
-        except asyncio.TimeoutError:
-            LOGGER.info("timeouted waiting for message for system: %s, account: %s", inventory_id, org_id)
-            PAIR_MISS.inc()
+        # RHSM systems do not need to wait for advisor message
+        if reporter == "rhsm-system-profile-bridge":
+            LOGGER.info("reporter %s received inventory message for system: %s, account: %s", reporter, inventory_id, org_id)
```

Comment:
The message should say that the waiting is skipped or something. And the inventory word in the msg shouldn't be hardcoded. Yes, we expect all of rhsm-system-profile-bridge messages from inventory, but technically they can come from advisor too, let's set "inventory" or "advisor" based on a condition, it'll be better for debugging purposes.

Reply:
How is it possible that they can come from advisor? I thought the rule based analysis cannot be done on the RSHM systems?

Reply:
You are right, we expect that RHSM systems are not processed by advisor and thus such messages **shouldn't** be sent. But since advisor is a black box here and this code handles messages from both inventory and advisor sources, I just wanted a clarity in the logged message (adding a condition like in other log messages in this function) to help debugging if anything goes wrong or changes on advisor side.